### PR TITLE
Add section describing "Aspects Under Test"

### PR DESCRIPTION
**Pull request by kasperisager**
I'm aiming to also add "Language", "Audio", and "Visuals" to the list of common aspects.

### CODE REVIEW

File: act-rules-format.bs [hunk 1]
```diff
@@ -83,58 +83,31 @@ Even within a rules group, certain rules may not be used. This leaves one fewer
...
 
 
-Test Subject Types (Input Data) {#input-types}
+Aspects Under Test {#input-aspects}
 ===================================
 
-Web pages, including publications and applications, go through many different stages before it is rendered to the end user. For example, PHP may be used to put various pieces of content into a template. That template is then sent as an HTML text document to a web browser. The browser in turn parses it and turns it into a DOM tree before rendering it to the screen. Accessibility tests could be run at each of these stages. It is therefore important to specify the *test subject type* that an ACT Rule expects.
+ACT Rules can operate on different aspects of the subject being tested. For example, rendering a web page to an end user involves multiple different technologies, some or all of which may be of interest to an ACT Rule. Some rules may need to operate directly on the HTTP messages exchanged between a server and a client, while others need to operate on the DOM tree exposed by a web browser. Some rules may even need to operate on several aspects simultaneously such as both the HTTP messages and the DOM tree.
```
Comment (WilcoFiers) [inline act-rules-format.bs]: This isn't actually explaining what an "aspect under test" **is**.

Comment (kasperisager) [reply act-rules-format.bs]: Good point!

**Comment by kasperisager** [comment]
@WilcoFiers I'm afraid I won't be able to fit this in today. Will see if I can get around to it Monday morning!

File: act-rules-format.bs [hunk 2]
```diff
@@ -83,58 +83,43 @@ Even within a rules group, certain rules may not be used. This leaves one fewer
...
-The following test subject types are common in accessibility testing:
-
-- HTTP Response
-- DOM Tree
-- Rendered Page
-- Template
-- Script
-
-Other types of test subjects may be possible. In those cases the ACT Rule MUST include a detailed description of the test subject type, or a reference to the that description.
+Each ACT Rule MUST include a description of all the aspects under test by the rule. Each aspect MUST be discrete and no overlap exist between aspects. Some aspects are already well defined within the context of web content, such as HTTP messages, DOM tree, and CSS styling, and do not warrant a detailed description. Other aspects may not be well defined or even specific to web content. In these cases, an ACT Rule SHOULD include a detailed description, or a reference to one, of the aspect in question.
```
Comment (maryjom) [inline act-rules-format.bs]: Suggested editorial change to the 2nd sentence: Each aspect MUST be discrete with no overlap between the aspects.

File: act-rules-format.bs [hunk 3]
```diff
@@ -83,58 +83,43 @@ Even within a rules group, certain rules may not be used. This leaves one fewer
...
 
+Common Aspects {#input-aspects-common}
+--------------------------------------
 
-HTTP Response Testing {#input-types-http-response}
---------------------------------------------
+### HTTP Messages ### {#input-aspects-http}
 
-Testing the files as they are served to the web browser (or other user agent) has its limitations. The files may be manipulated in different ways through presentation and scripting. Although this is an excellent place for parser testing.
+The HTTP messages exchanged between a client and a server as part of requesting a web page may be of interest to ACT Rules that wish to perform validation of e.g. HTTP headers or unparsed HTML and CSS.
```
Comment (maryjom) [inline act-rules-format.bs]: This sentence seems to be missing something or reads a little awkwardly.  Is it supposed to say, "...ACT Rules which perform validation, such as HTTP headers or unparsed HTML and CSS.

File: act-rules-format.bs [hunk 4]
```diff
@@ -83,58 +83,43 @@ Even within a rules group, certain rules may not be used. This leaves one fewer
...
 
+### DOM Tree ### {#input-aspects-dom}
 
-DOM Tree Testing {#input-types-dom-tree}
------------------------------------
+The DOM tree constructed from parsing HTML, and optionally executing DOM manipulating JavaScript, may be of interest to ACT Rules that wish to test the structure of web pages. In the DOM tree, information about individual elements of a web page, and their relations, becomes available.
```
Comment (maryjom) [inline act-rules-format.bs]: I would remove the phrase "wish to" from the first sentence.  The rules don't "wish". ;)

Also to be more clear the last sentence suggest: "...information about individual elements of a web page, and the relationships between them, becomes available."  Or am I missing the correct meaning?

File: act-rules-format.bs [hunk 5]
```diff
@@ -83,58 +83,43 @@ Even within a rules group, certain rules may not be used. This leaves one fewer
...
 
-After the web browser (or other user agent) has parsed the files, and executed scripts to get it into a specific state (be it the initial state or any other), tests can be run against the DOM Tree. The DOM Tree can be tested for things like correct parent-child relationships, use of required attributes or properties and more.
+The means by which the DOM tree is constructed, be it by a web browser or not, is not of importance as long as the construction follows any applicable specifications that might exist, such as [[DOM]].
 
+### CSS Styling ### {#input-aspects-css}
 
-Rendered Page Testing {#input-types-rendered-page}
---------------------------------------------
+The computed CSS styling resulting from parsing CSS and applying it to the DOM may be of interest to ACT Rules that wish to test the web page as presented to the user. Through CSS styling, information about the position, the fore- and background colors, the visibility, and more, of elements becomes available.
```
Comment (maryjom) [inline act-rules-format.bs]: Again, I would remove "wish to" and use the full word "foreground" rather than "fore-".

File: act-rules-format.bs [hunk 6]
```diff
@@ -83,58 +83,43 @@ Even within a rules group, certain rules may not be used. This leaves one fewer
...
 
-A template is a document that has "open" fields that are filled with pieces of content or other templates. E.g. an HTML template contains a head with a variable title, a predefined navigation, and a variable content area.
+The means by which the accessibility tree is constructed, be it by a web browser or not, is not of importance as long as the construction follows any applicable specifications that might exist, such as [[CORE-AAM-1.1]].
 
-<div class=example>
-`<img>` tags with a variable `src`, MUST NOT have a static `alt` value.
-</div>
+### Language ### {#input-aspects-text}
 
+Language, either written or spoken, as available in e.g. nodes of the DOM or accessibility trees may be of interest to ACT Rules that wish to test things like complexity or intention of the language. For example, an ACT Rule might wish to test that certain paragraphs of text within the DOM tree do not exceed a certain readability score or that the text alternative of an image provides a sufficient description thereof.
```
Comment (maryjom) [inline act-rules-format.bs]: The "e.g." in the middle of the first sentence makes it read a bit awkwardly.
Again, the use of "wish to".  And just wondering, how do you test the "intention" of the language?  How do you test intent?

File: act-rules-format.bs (hunk 2 above)
Comment (moekraft) [reply act-rules-format.bs]: I agree with Mary Jo's editorial change. I'm not loving the term aspect but other than that I think the update is good. That said, I'm having a hard time finding an alternative. Some synonyms include feature, condition, facet, and part.

File: act-rules-format.bs (hunk 3 above)
Comment (WilcoFiers) [reply act-rules-format.bs]: of interest to ACT Rules. For example to perform validation of HTTP headers or unparsed HTML and CSS.

File: act-rules-format.bs [hunk 7]
```diff
@@ -83,58 +83,43 @@ Even within a rules group, certain rules may not be used. This leaves one fewer
...
 
-Web pages, including publications and applications, go through many different stages before it is rendered to the end user. For example, PHP may be used to put various pieces of content into a template. That template is then sent as an HTML text document to a web browser. The browser in turn parses it and turns it into a DOM tree before rendering it to the screen. Accessibility tests could be run at each of these stages. It is therefore important to specify the *test subject type* that an ACT Rule expects.
+ACT Rules can operate on different aspects of the subject being tested. An aspect is a distinct part of the test subject itself or its underlying implementation. For example, rendering a web page to an end user involves multiple different technologies, some or all of which may be of interest to an ACT Rule. Some rules may need to operate directly on the HTTP messages exchanged between a server and a client, while others need to operate on the DOM tree exposed by a web browser. Some rules may even need to operate on several aspects simultaneously such as both the HTTP messages and the DOM tree.
 
-The following test subject types are common in accessibility testing:
```
Comment (WilcoFiers) [inline act-rules-format.bs]: Should add an example below this.

**Pull request closed without merging**